src/components/Board/TileEditor/TileEditor.component.js

```diff
@@ -1,711 +1,741 @@
-    if (this.editingTile()) {
-      let imageUploadedDataArray = new Array(this.state.editingTiles.length);
-      imageUploadedDataArray.fill(this.defaultimageUploadedData);
-      this.setState({ imageUploadedData: imageUploadedDataArray });
-    } else {
-      this.setState({
-        imageUploadedData: new Array(this.defaultimageUploadedData)
+      this.setState({ tile, linkedBoard: '' });
+    };
+
+    handleBack = event => {
+      this.setState({ activeStep: this.state.activeStep - 1 }, () => {
+        this.setLinkedBoard();
       });
-    }
-  }
+      this.setState({ selectedBackgroundColor: '' });
+      this.setState({ isEditImageBtnActive: false });
+    };
 
-  handleInputImageChange = (blob, fileName, blobHQ) => {
-    if (!this.state.imageUploadedData.length) {
-      this.createimageUploadedDataArray();
-    }
-    this.setimageUploadedData(true, fileName, blobHQ, blob);
-    this.setState({ isEditImageBtnActive: true });
-    const image = URL.createObjectURL(blob);
-    this.updateTileProperty('image', image);
-  };
-
-  handleLoadingStateChange = isLoading => {
-    this.setState({ isLoading: isLoading });
-  };
-
-  setimageUploadedData = (isUploaded, fileName, blobHQ = null, blob = null) => {
-    const { activeStep } = this.state;
-    let imageUploadedData = this.state.imageUploadedData.map((item, indx) => {
-      if (indx === activeStep) {
-        return {
-          ...item,
-          isUploaded: isUploaded,
-          fileName: fileName,
-          blobHQ: blobHQ,
-          blob: blob
-        };
+    handleNext = async event => {
+      this.setState({ activeStep: this.state.activeStep + 1 }, () => {
+        this.setLinkedBoard();
+      });
+      this.setState({ selectedBackgroundColor: '' });
+      this.setState({ isEditImageBtnActive: false });
+    };
+
+    handleSearchClick = (event, currentLabel) => {
+      this.setState({ isSymbolSearchOpen: true, autoFill: currentLabel || '' });
+      this.setState({ isEditImageBtnActive: false });
+    };
+
+    handleColorChange = event => {
+      const color = event ? event.target.value : '';
+      this.setState({ selectedBackgroundColor: color });
+      if (event) {
+        this.updateTileProperty('backgroundColor', event.target.value);
       } else {
-        return item;
+        this.updateTileProperty('backgroundColor', this.getDefaultColor());
       }
-    });
-    this.setState({ imageUploadedData: imageUploadedData });
-  };
-
-  handleSymbolSearchChange = ({ image, labelKey, label, keyPath }) => {
-    return new Promise(resolve => {
-      this.updateTileProperty('labelKey', labelKey);
-      this.updateTileProperty('label', label);
-      this.updateTileProperty('image', image);
-      if (keyPath) this.updateTileProperty('keyPath', keyPath);
-      if (this.state.imageUploadedData.length) {
-        this.setimageUploadedData(false, '');
+    };
+
+    handleChangeComplete = color => {
+      const colorhex = color.hex ? color.hex : '';
```

Comment:
Is this safe? 

We check whether `color` is truthy on line 414; I assume it's possible for it to be undefined so we could get a TypeError here. Perhaps we could move this inside the if statement or use optional chaining?

```suggestion
      const colorhex = color?.hex || '';
```

Reply:
Sure , will make this change

Reply:
I have made the changes @sylvansson

Reply:
Thank you 🙏 I'll let one of the core maintainers review your changes :-)
@martinbedouret @RodriSanchez1 @tomivm